sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -505,26 +519,53 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
 
     task_refs = []
     templates = []
+    cel_conditions = {}
+    condition_when_refs = {}
     condition_task_refs = {}
+    string_condition_refs = {}
     for template in raw_templates:
       if template['kind'] == 'Condition':
-        condition_task_ref = [{
-            'name': template['metadata']['name'],
-            'params': [{
-                'name': p['name'],
-                'value': p.get('value', '')
-              } for p in template['spec'].get('params', [])
-            ],
-            'taskSpec': _get_super_condition_template(),
-        }]
-        condition_refs[template['metadata']['name']] = [
-            {
-              'input': '$(tasks.%s.results.status)' % template['metadata']['name'],
-              'operator': 'in',
-              'values': ['true']
-            }
-          ]
-        condition_task_refs[template['metadata']['name']] = condition_task_ref
+        if DISABLE_CEL_CONDITION:
+          condition_task_spec = _get_super_condition_template()
+        else:
+          condition_task_spec = _get_cel_condition_template()
+
+        condition_params = template['spec'].get('params', [])
+        if condition_params:
+          condition_task_ref = [{
+              'name': template['metadata']['name'],
+              'params': [{
+                  'name': p['name'],
+                  'value': p.get('value', '')
+                } for p in template['spec'].get('params', [])
+              ],
+
+              'taskSpec' if DISABLE_CEL_CONDITION else 'taskRef': condition_task_spec
+          }]
+          condition_refs[template['metadata']['name']] = [
+              {
+                'input': '$(tasks.%s.results.status)' % template['metadata']['name'],
+                'operator': 'in',
+                'values': ['true']
+              }
+            ]
+          # Don't use additional task if it's only doing literal string == and !=
```

Comment:
That's actually misleading, as it breaks the 1:1 relationship between DSL and Tekton entities. `dsl.Condition` is expected to generate a single task or custom task, if it generates it only for some operators, tracking becomes very unobvious.

```python
with dsl.Condition(op.['results_len'] > 0): # custom task is generated
  ...


with dsl.Condition(op.output['status'] == "succeeded"):  # custom task is not generated
  ...
```

Reply:
done